File: MatlabOctaveCodeSnippets/snip_systems_SOS_decomposition.m
B=[0.6 0.1 2.3 4.5]; %numerator
A=[1 -1.2 0.4 5.2 1.1]; %denominator 
[z,p,k]=tf2zp(B,A); %convert to zero-pole for improved 
[sos,g] = zp2sos(z, p, k) %to SOS, g is the overall gain

File: PythonCodeSnippets/snip_systems_SOS_decomposition.py
import numpy as np
from scipy import signal

B = np.array([0.6, 0.1, 2.3, 4.5])  # numerator
A = np.array([1, -1.2, 0.4, 5.2, 1.1])  # denominator

z, p, k = signal.tf2zpk(B, A)  # convert to zero-pole
sos = signal.zpk2sos(z, p, k)  # to SOS

print("Zeros:", z)
print("Poles:", p)
print("Gain:", k)
print("Second-Order Sections:", sos)
